packages/transactions/src/types.ts

```diff
@@ -273,7 +273,8 @@ export function serializeMemoString(memoString: MemoString): Uint8Array {
 }
 
 export function deserializeMemoString(bytesReader: BytesReader): MemoString {
-  const content = bytesToUtf8(bytesReader.readBytes(MEMO_MAX_LENGTH_BYTES));
+  let content = bytesToUtf8(bytesReader.readBytes(MEMO_MAX_LENGTH_BYTES));
+  content = content.replace(/\u0000*$/, ''); // remove all trailing unicode empty characters
```

Comment:
I'm curious if this always trims as expected. I think it depends on how `bytesToUtf8` handles invalid utf8 bytes. For example, I know nodejs encoders typically outputs the "unicode replacement character" `U+FFFD`.

In other words, it seems possible to construct a tx with a memo that has equality with another memo even with different bytes.

It might be helpful to add test vectors with 
A) Last character is a zero byte (`0x00`)
B) Trailing bytes are invalid utf8

Reply:
Yeah, hard to say what the best behavior is here. I'm thinking it would be fine to leave as is, but some users might not expect the right-padded zero bytes from a serialized transaction. So all trailing zero-bytes are trimmed (comment might have been confusing). `\u0000` is ASCII/UTF-8 compatible `0` byte.
B) is interesting.
Added more test-cases.

But essentially, it's always possible to break this -- if this behavior is too unclear, we can also leave as is and let the developer deal with `.trim()`

Reply:
Test cases LGTM

Reply:
Where are these null characters coming from? Rather than removing them, is it possible not to introduce them in the first place? I think if the user intentionally puts null characters in this field, they should be preserved. There are legitimate use-cases for this. For example, maybe the user wants to put binary data in the field, or maybe they have a list of strings and are using `\0` as a separator

Reply:
That won't be possible, while adding this feature.

The null characters come from the serialization on the blockchain (wire format). The memo has a fixed length of bytes, and is right-padded if the actual content is shorter.

I'll double check if this is true in the impl, or if there's a way to strip the trailing bytes, hard to follow the macro magic 😅

Reply:
I can't find anything on `memo` in the SIPs, but core uses `read_exact` via `impl_byte_array_message_codec` for it, on a fixed byte length of 34. So I'm assuming this was always how it worked.

Reply:
So if it's a fixed-length field, we'd still submitting the tx with the null bytes then? Maybe it is better to just make that explicit to the user, I don't know

Reply:
Exchanges pretty much all use null terminated strings for memo, so this PR makes sense